src/v/container/include/container/fragmented_vector.h

```diff
@@ -70,10 +70,15 @@ class fragmented_vector {
 
 public:
     using this_type = fragmented_vector<T, fragment_size_bytes>;
+    using backing_type = std::vector<std::vector<T>>;
     using value_type = T;
     using reference = T&;
     using const_reference = const T&;
     using size_type = size_t;
+    using allocator_type = backing_type::allocator_type;
+    using difference_type = backing_type::difference_type;
+    using pointer = backing_type::pointer;
+    using const_pointer = backing_type::const_pointer;
```

Comment:
Why is this the difference/pointer type of `backing_type` and not `backing_type::value_type`?

Reply:
Yeah I guess it should be what you say. The allocator business always makes this complicated.

This however made me realize that I missed a thing in my patch to unordered_dense and we currently wouldn't use our custom type for the bucket array (but their segmented_vector) so I will need to fix that first.

Reply:
Fixed now